tests/models/test_requests.py

```diff
@@ -226,3 +226,27 @@ def content() -> typing.Iterator[bytes]:
     request.read()
     pickle_request = pickle.loads(pickle.dumps(request))
     assert pickle_request.content == b"test 123"
+
+
+def test_with_params():
+    request = httpx.Request("GET", "http://example.com", params={"a": "1", "b": "2"})
+    assert str(request.url) == "http://example.com?a=1&b=2"
+
+
+def test_merge_existing_params():
+    request = httpx.Request(
+        "GET", "http://example.com?c=3", params={"a": "1", "b": "2"}
+    )
+    assert str(request.url) == "http://example.com?c=3&a=1&b=2"
```

Comment:
This is the only one that fails on main, right?

Reply:
let me test.

Reply:
yes, and `test_empty_params` too.

Reply:
yes, and `test_empty_params` too.

Reply:
can you just leave the tests that actually break on main on this PR?

Reply:
fine

Reply:
thanks